server/utils/utils.go

```diff
@@ -65,3 +68,41 @@ func IsJiraCloudURL(jiraURL string) (bool, error) {
 	}
 	return strings.HasSuffix(u.Hostname(), ".atlassian.net"), nil
 }
+
+type JiraStatus struct {
+	State string `json:"state"`
+}
+
+// IsJiraAccessible checks if `/status` endpoint of the Jira URL is accessible
+// and responding with the correct state which is "RUNNING"
+func IsJiraAccessible(jiraURL string) (bool, error) {
+	u, err := url.Parse(jiraURL)
+	if err != nil {
+		return false, nil
+	}
+	if u.Host == "" {
+		return false, errors.Errorf("Invalid URL, no hostname: %q", jiraURL)
```

Comment:
Seems odd that this function has an error return value, but is only returning an error in one case. For instance for the `r.StatusCode != http.StatusOK` block, we could return to the admin that their `Jira server returned http status code 404 when checking for availability`.

Reply:
added that one, and also added below one;

`Jira server is not in correct state, it should be up and running`